spec/bson/byte_buffer_spec.rb

```diff
@@ -30,16 +30,37 @@
       let(:buffer) do
         described_class.new
       end
-
-      before do
-        buffer.put_int32(5)
+      context '#put_int32' do 
+        before do
+          buffer.put_int32(5)
+        end
+
+        it 'returns the length of the buffer' do
+          expect(buffer.length).to eq(4)
+        end
       end
 
-      it 'returns the length of the buffer' do
-        expect(buffer.length).to eq(4)
+      context '#put_uint32' do 
+        context 'when number is in range' do 
+          before do
+            buffer.put_uint32(5)
+          end
+
+          it 'returns the length of the buffer' do
+            expect(buffer.length).to eq(4)
+          end
+        end
+
+        context 'when number is not in range' do 
+          it 'raises error on out of range number' do
+            expect{ buffer.put_uint32(4294967296) }.to raise_error(RangeError)
+            expect{ buffer.put_uint32(-4294967296) }.to raise_error(RangeError)
```

Comment:
Each of these assertions should be in its own example, and the negative test should use -1 as the value closest to the valid range.

Reply:
-1 is not out of the range... It automatically gets converted into an unsigned integer (4294967295 to be specific), that's why I put -4294967296 as the lower range.

Reply:
-1 is not an unsigned integer, therefore my first thought was it should raise an error. @egiurleo thoughts?

Reply:
I think I agree with Oleg here. If the user wants to store 4294967295, they should specify 4294967295 rather than -1. Of course those numbers are the same as far as the byte buffer is concerned, but to the user they are different numbers and I believe our API should take that into account.

Reply:
Got it!